core/src/main/java/net/opentsdb/query/processor/groupby/GroupByNumericArrayIterator.java

```diff
@@ -166,19 +167,21 @@ public GroupByNumericArrayIterator(
     }
 
     // TODO: Need to check if it makes sense to make this threshold configurable
-    if (sources.size() >= NUM_THREADS) {
-      NumericArrayAggregator[] valuesCombiner = new NumericArrayAggregator[NUM_THREADS];
-      for (int i = 0; i < valuesCombiner.length; i++) {
-        valuesCombiner[i] = createAggregator(node, factory, size);
-      }
-
-      accumulateInParallel(sources, valuesCombiner);
-
-    } else {
-      for (final TimeSeries source : sources) {
+    NumericArrayAggregator[] valuesCombiner = new NumericArrayAggregator[NUM_THREADS];
+    for (int i = 0; i < valuesCombiner.length; i++) {
+      valuesCombiner[i] = createAggregator(node, factory, size);
+    }
+    
+    if (size == 0 || !(sources.iterator()
+        .next() instanceof Downsample.DownsampleResult.DownsampleTimeSeries)) {
```

Comment:
NPE here if the sources iterator is null or doesn't have a next. e.g. serdes <-- gb <-- time series source

Reply:
good point, will take of it.

Reply:
NVM, you were right, this is the time series sources, not graph source.